analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java

```diff
@@ -102,14 +107,50 @@ public void run()
 		queryOptions.setAdapter(featureAdapterIds);
 
 		// This is required due to some funkiness in GeoWaveInputFormat
-		queryOptions.getAdaptersArray(inputDataStore.createAdapterStore());
+		AdapterStore adapterStore = inputDataStore.createAdapterStore();
+		queryOptions.getAdaptersArray(adapterStore);
 
 		// Add a spatial filter if requested
 		DistributableQuery query = null;
-		if (bbox != null) {
-			Geometry geom = new GeometryFactory().toGeometry(bbox);
-			query = new SpatialQuery(
-					geom);
+		try {
+			if (cqlFilter != null) {
+				Geometry bbox = null;
+				ByteArrayId cqlAdapterId;
+				if (adapterId == null) {
+					cqlAdapterId = featureAdapterIds.get(0);
+				}
+				else {
+					cqlAdapterId = new ByteArrayId(
+							adapterId);
+				}
+
+				DataAdapter adapter = adapterStore.getAdapter(cqlAdapterId);
+
+				if (adapter instanceof FeatureDataAdapter) {
+					final String geometryAttribute = ((FeatureDataAdapter) adapter)
+							.getFeatureType()
+							.getGeometryDescriptor()
+							.getLocalName();
+					Filter filter;
+					filter = ECQL.toFilter(cqlFilter);
+
+					final ExtractGeometryFilterVisitorResult geoAndCompareOpData = (ExtractGeometryFilterVisitorResult) filter
+							.accept(
+									new ExtractGeometryFilterVisitor(
+											GeoWaveGTDataStore.DEFAULT_CRS,
+											geometryAttribute),
+									null);
+					bbox = geoAndCompareOpData.getGeometry();
+				}
+
+				if ((bbox != null) && !bbox.equals(GeometryUtils.infinity())) {
+					query = new SpatialQuery(
+							bbox);
+				}
+			}
+		}
+		catch (CQLException e) {
+			LOGGER.error("Unable to parse CQL: " + cqlFilter);
```

Comment:
ideally we'd have a stack trace (ie. add `, e);`  Not worth disapproving the PR jsut for that.